src/commands/kv/bucket/mod.rs

```diff
@@ -100,22 +98,26 @@ fn generate_url_safe_path(path: &Path, directory: &Path) -> Result<String, failu
     Ok(path.to_string())
 }
 
-// Appends the SHA-256 hash of the path's file contents to the url-safe path of a file to
+// Adds the SHA-256 hash of the path's file contents to the url-safe path of a file to
 // generate a versioned key for the file and its contents. Returns the url-safe path prefix
 // for the key, as well as the key with hash appended.
-// e.g (sitemap.xml, sitemap.xml-ec717eb2131fdd4fff803b851d2aa5b1dc3e0af36bc3c8c40f2095c747e80d1e)
-pub fn generate_url_safe_key_and_hash(
+// e.g (sitemap.xml, sitemap.ec717eb2131fdd4fff803b851d2aa5b1dc3e0af36bc3c8c40f2095c747e80d1e.xml)
+pub fn generate_path_and_key(
     path: &Path,
     directory: &Path,
     value: Option<String>,
 ) -> Result<(String, String), failure::Error> {
-    let url_safe_path = generate_url_safe_path(path, directory)?;
+    // strip the bucket directory from both paths for ease of reference.
+    let relative_path = path.strip_prefix(directory).unwrap();
```

Comment:
can we do an if let with a failure in the else

Reply:
we can; fwiw this is code moved from a lower function up into this one verbatim, so it's sort of status quo

Reply:
ok :)